vscode-dotnet-runtime-library/src/Utils/WebRequestWorker.ts

```diff
@@ -73,12 +77,50 @@ export class WebRequestWorker
             }
         });
 
-            this.client = setupCache(uncachedAxiosClient,
-                {
-                    storage: buildMemoryStorage(),
-                    ttl: this.cacheTimeToLive
-                }
-            );
+        // Record when the web requests begin:
+        // Register this so it happens before the cache https://axios-cache-interceptor.js.org/guide/interceptors#explanation
+        uncachedAxiosClient.interceptors.request.use( req => {
+            req.headers.startTime = process.hrtime.bigint();
+            return req;
+        });
+
+        // Record when the server responds:
+        // Register this so it happens after the cache -- this means we need to check if the result is cached before reporting perf data!
+        // ^ the request should not be processed if it is in the cache, it seems nonsensical to check this before the cache is hit even though this is possible
+        uncachedAxiosClient.interceptors.response.use(res =>
```

Comment:
IIRC this will only handle succeeded requests (the 'resolve' delegate) - I think you'd also need to specify a delegate for the 'faulted' promise delegate if we needed to trace failed responses.

Reply:
That's a great point. We have our own timeout abort controller, but I'm not sure if that will handle responses that get rejected with something like 401/404. So, I should write a handler for that.

Reply:
This is a great of example of why its worth reading everything in the medium article even if some of it is boring and just explains the code verbatim. Thank you!